crates/nu-parser/src/parser.rs

```diff
@@ -1105,20 +1119,21 @@ pub fn parse_binary(
                 }
             }
 
-            if hex_value.len() % 2 != 0 {
-                return (
-                    garbage(span),
-                    Some(ParseError::IncorrectValue(
-                        "incomplete binary".into(),
-                        span,
-                        "number of binary digits needs to be a multiple of 2".into(),
-                    )),
-                );
+            let required_padding = (min_digits_per_byte - binary_value.len() % min_digits_per_byte)
+                % min_digits_per_byte;
+
+            if required_padding != 0 {
+                binary_value = {
+                    let mut tail = binary_value;
+                    let mut binary_value: Vec<u8> = vec![b'0'; required_padding];
+                    binary_value.append(&mut tail);
+                    binary_value
+                };
             }
 
-            let str = String::from_utf8_lossy(&hex_value).to_string();
+            let str = String::from_utf8_lossy(&binary_value).to_string();
 
-            match decode_hex(&str) {
+            match decode_with_base(&str, base, min_digits_per_byte) {
```

Comment:
Interesting and sad that the error message thrown by that get's eaten with the eager assumption that everything unknown could be an external command. (I know that it is a valid filename)

Reply:
@sholderbach - for which case? I don't quite follow

Reply:
Try to put invalid content in the literal body. E.g. `0b[ff]` I got a unknown executable error

Reply:
Oooooh, hmm, let me think about this. We're going to have this issue generally, not just with the binary literals.

I think we should have some logic that says "if it starts with 0x[" or "if it starts with 0b[" this it's definitely a binary literal and if it doesn't parse, we should error there.